executor/union_scan.go

```diff
@@ -73,13 +49,36 @@ func (udb *DirtyDB) GetDirtyTable(tid int64) *DirtyTable {
 
 // DirtyTable stores uncommitted write operation for a transaction.
 type DirtyTable struct {
+	tid int64
 	// addedRows ...
 	// the key is handle.
-	addedRows   map[int64][]types.Datum
+	addedRows   map[int64]struct{}
 	deletedRows map[int64]struct{}
 	truncated   bool
 }
 
+// AddRow adds a row to the DirtyDB.
+func (dt *DirtyTable) AddRow(handle int64, row []types.Datum) {
+	for i := range row {
+		if row[i].Kind() == types.KindString {
+			row[i].SetBytes(row[i].GetBytes())
```

Comment:
Why do we need to do this change?

Reply:
It is moved from https://github.com/pingcap/tidb/pull/9272/files#diff-2f7dd37a7b77b4ed33d6b9c88cd69f3bL41 .
I've made the dirty table could add/delete row itself and add the tid for it, which makes it more cohesion. Then it could be used in `buildAndSortAddedRows`.

Reply:
I know, I just don't quite understand this logic.

Reply:
I try to remove this logic, and it can run all unit tests. Could we remove it?

Reply:
I try to remove this logic, and it can run all unit tests. Could we remove it?

Reply:
OK, they should be useless now.